generate.py

```diff
@@ -226,18 +252,18 @@ def fix(obj):
 filtered_api_methods.update(add)
 filtered_api_methods = dict(sorted(filtered_api_methods.items(), key=lambda x: x[0][:-1]))
 
-#pprint.pprint(filtered_api_methods, sort_dicts=False)
+# pprint.pprint(filtered_api_methods, sort_dicts=False)
 
 # Write the argument models Python module.
-with open('toshling/models/argument_types.py', 'w') as f:
-    arguments = (api_method['argument'] for api_method in filtered_api_methods.values() if api_method['argument'])
-    f.write(statham.serializers.python.serialize_python(*arguments))
+arguments = (api_method['argument'] for api_method in filtered_api_methods.values() if api_method['argument'])
+arg_types_path = Path('toshling/models/argument_types.py')
+arg_types_path.write_text(statham.serializers.python.serialize_python(*arguments))
 
 # Automatically generate Python code for the endpoints.
-classes = []
+classes: List[Dict[str, List[Dict[str, Any]]]] = []
 subclasses = []
 prev_length = 0
-for crumbs, api_method in reversed(filtered_api_methods.items()):
+for crumbs, api_method in sorted(filtered_api_methods.items(), reverse=True):
```

Comment:
`filtered_api_methods` is already sorted above. `dict`s for Python 3.7+ are ordered, so it should be all good. If we want to make the code generation compatible with older versions, we should look at the other places `filtered_api_methods` is used and sort them, too.

Reply:
It actually only appears twice.

Actually, in the other place when it's used (the classes), in my current code I do a different sorting: Alphabetically while preserving dependencies

```python
    def key_sort_class_name(dict_class: Dict[str, Any]) -> List[str]:
        """Sort Classes alphabetically, with dependance on sub-classes"""
        crumbs = re.findall(r"[A-Z][a-z0-9]+", dict_class["name"])
        return crumbs if len(crumbs) > 1 else [crumbs[0], "zzz"]

    end_tmp = Path('_endpoints.py.j2')
    template = Template(end_tmp.read_text())
    end_out = Path('toshling/_endpoints.py')

    classes = get_classes(filtered_api_methods)
    classes_sorted = sorted(classes.values(), key=key_sort_class_name)
```

About order preserved in dict, it is true indeed in Python 3.7+. I noticed it but was not sure it was an official feature.
We should not care about support of older versions since we plan to add `Dataclasses` which are also Python 3.7+.

But I believe it's nice to specifically do the sorting: code is easier to read.

Reply:
That's my point, though - `filtered_api_methods` is specifically sorted [here](https://github.com/ioces/toshling/blob/master/generate.py#L227) using a particular key, then that sorting is maintained while writing `argument_types.py`, as well as generating the classes.

I'm happy to have sorting implemented on the fly wherever needed, but we ought to be consistent.

Reply:
Indeed, but

1. I have left the original dictionary sorted
2. I have an error with `reversed` so that's why I used `sorted`  (Python 3.7.1 32-bits)

```
Traceback (most recent call last):
  File ".\generate.py", line 264, in <module>
    for crumbs, api_method in reversed(filtered_api_methods.items()):
TypeError: 'dict_items' object is not reversible
```

![image](https://user-images.githubusercontent.com/1401809/129829952-7076eb95-db69-4e69-ad3a-11b9b61c32b0.png)

Maybe we could just sort when doing a reserved sorting of items and store it (instead of sorting of dictionary)?